hazelcast/src/main/java/com/hazelcast/internal/util/futures/ChainingFuture.java

```diff
@@ -57,7 +59,7 @@ private void registerCallback(InternalCompletableFuture<T> future,
                     completeExceptionally(t);
                 }
             }
-        });
+        }, CALLER_RUNS);
```

Comment:
There is a problem with `CALLER_RUNS` here, it leads to `IllegalThreadStateException`

Reply:
I'm thinking of adding a new executor to `ConcurrencyUtil`:
`public static final Executor INTERNAL_EXECUTOR = Executors.newWorkStealingPool();`
to use it whenever we can't use the `CALLER_RUNS` and have no access to the `internalAsyncExecutor`.

What do you think?

Reply:
I think common-FJP can also be used here. This code is mostly triggered by user actions.